homeassistant/components/zha/__init__.py

```diff
@@ -321,19 +316,30 @@ def __init__(self, endpoint, in_clusters, out_clusters, manufacturer,
                 kwargs.get('entity_suffix', ''),
             )
 
-        for cluster_id, cluster in in_clusters.items():
-            cluster.add_listener(self._in_listeners.get(cluster_id, self))
-        for cluster_id, cluster in out_clusters.items():
-            cluster.add_listener(self._out_listeners.get(cluster_id, self))
-
         self._endpoint = endpoint
         self._in_clusters = in_clusters
         self._out_clusters = out_clusters
         self._state = ha_const.STATE_UNKNOWN
         self._unique_id = unique_id
 
+        # Normally the entity itself is the listener. Sub-classes may set this
+        # to a dict of cluster ID -> listener to receive messages for specific
+        # clusters separately
+        self._in_listeners = {}
+        self._out_listeners = {}
```

Comment:
This will reset the listeners added in the child. You can init the parent before the setting the rest of the child's attributes.